guides/source/active_support_instrumentation.md

```diff
@@ -3,623 +3,1160 @@
 
 ### Action Cable
 
 #### `perform_action.action_cable`
 
-| Key              | Value                     |
-| ---------------- | ------------------------- |
-| `:channel_class` | Name of the channel class |
-| `:action`        | The action                |
-| `:data`          | A hash of data            |
+The event is emitted when a channel action is invoked.
+
+For example:
+
+```ruby
+# Client sends: { "action":"speak", "message":"Hi" }
+class ChatChannel < ApplicationCable::Channel
+  def speak(data) = Message.create!(text: data["message"])
+end
+```
+
+The event payload (`event.payload`) includes the following keys (with typical
+example values).
+
+| Payload Key      | Description               | Example               |
+| ---------------- | ------------------------- | --------------------- |
+| `:channel_class` | Name of the channel class | `"ChatChannel"`       |
+| `:action`        | The action                | `"speak"`             |
+| `:data`          | A hash of data            | `{ "message"=>"Hi" }` |
 
 #### `transmit.action_cable`
 
-| Key              | Value                     |
-| ---------------- | ------------------------- |
-| `:channel_class` | Name of the channel class |
-| `:data`          | A hash of data            |
-| `:via`           | Via                       |
+The event is emitted when a message is transmitted from a channel.
+
+For example:
+
+```ruby
+class ChatChannel < ApplicationCable::Channel
+  def subscribed
+    transmit(ok: true)
+  end
+end
+```
+
+The event payload (`event.payload`) includes the following keys (with typical
+example values).
+
+| Payload Key      | Description               | Example          |
+| ---------------- | ------------------------- | ---------------- |
+| `:channel_class` | Name of the channel class | `"ChatChannel"`  |
+| `:data`          | A hash of data            | `{ "ok"=>true }` |
+| `:via`           | Via                       | `"websocket"`    |
 
 #### `transmit_subscription_confirmation.action_cable`
 
-| Key              | Value                     |
-| ---------------- | ------------------------- |
-| `:channel_class` | Name of the channel class |
+The event is emitted when a subscription confirmation is sent to a client.
+
+For example:
+
+```ruby
+# Successful subscription auto-confirms -> event fires
+```
+
+The event payload (`event.payload`) includes the following keys (with typical
+example values).
+
+| Payload Key      | Description               | Example         |
+| ---------------- | ------------------------- | --------------- |
+| `:channel_class` | Name of the channel class | `"ChatChannel"` |
 
 #### `transmit_subscription_rejection.action_cable`
 
-| Key              | Value                     |
-| ---------------- | ------------------------- |
-| `:channel_class` | Name of the channel class |
+The event is emitted when a subscription rejection is sent to a client.
+
+For example:
+
+```ruby
+def subscribed
+  reject
+end
```

Comment:
We already show in what class `subscribed` is defined in the example above, but I think it's clearer if we use it for other examples as well:
```suggestion
class ChatChannel < ApplicationCable::Channel
  def subscribed
    reject
  end
end
```